probe-rs/src/architecture/xtensa/communication_interface.rs

```diff
@@ -720,6 +731,25 @@ impl MemoryInterface for XtensaCommunicationInterface {
 pub trait TypedRegister {
     fn register() -> Register;
     fn from_u32(value: u32) -> Self;
+    fn as_u32(&self) -> u32;
+}
+
+macro_rules! u32_register {
```

Comment:
We already have 3 traits for this. Maybe we should unify this in a later PR :)

Reply:
Baby steps :)

Reply:
Though I'm not even sure what the other 2 traits are. I can see MemoryMappedRegister (which don't apply, Xtensa doesn't really have the concept), but not sure about the third one.

Reply:
I am not sure we have 3. But I thought we have a couple of register traits. But some of them are for Arm core registers which have different requirements ofc :)